server/src/main/java/io/crate/metadata/doc/DocTableInfo.java

```diff
@@ -510,13 +511,21 @@ public Collection<ColumnIdent> notNullColumns() {
      * <ul>
      *  <li>OBJECT (IGNORED) sub-columns</li>
      *  <li>Empty arrays, or arrays with only null values</li>
+     *  <li>Internal object keys of the geo shape column, such as "coordinates", "type"</li>
      * </ul>
      */
     public Function<String, String> lookupNameBySourceKey() {
         if (versionCreated.onOrAfter(Version.V_5_5_0)) {
             return oidOrName -> {
-                String leafName = leafNamesByOid.get(oidOrName);
-                return leafName != null ? leafName : oidOrName;
+                if (oidOrName.startsWith(UNKNOWN_COLUMN_PREFIX)) {
+                    assert oidOrName.length() >= UNKNOWN_COLUMN_PREFIX.length() + 1 : "Column name must consist of at least one character";
```

Comment:
I initially had `startsWith("_")` as a micro-optimization. 

After running into geo shape issue (see second commit) I'm afraid that we might have (now or in the future) some complex objects which are serialised internally with underscore. 

For example, geo shape could have been serialised like `_coordinates`  - maybe let's compare extra 2 chars and be 100% safe that we don't mess ignored columns with some complex object structures.